### PR TITLE
Record performance numbers during benchmark runs

### PR DESCRIPTION
**Pull request by cirego**
This PR does a few things and the main purpose is to make it easier / faster to identify how quickly a benchmark ran and point the user at the correct Grafana dashboard / times.

The summary of changes in this PR are:

- Automatically generating link to Grafana centered on the time when the benchmark ran
- Add option to print logs from a container aftering waiting on exit
- Update the README with the new instructions for running this benchmark

This code, while currently in the `avro-upsert` benchmark directory, is planned for being used during the `chbench` benchmarks as well.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/5637)

### CODE REVIEW

File: test/benchmark/avro-upsert/metric-verifier/wait_metric [hunk 1]
```diff
@@ -10,32 +10,63 @@
...
+def print_dashboard_url(args: argparse.Namespace, start: datetime.datetime):
+    """Print the URL to see the metrics for this test"""
 
+    # Create parameters to see a dashboard with the metrics from this benchmark run
+    # Convert start and end to timestamps, with a little padding to make the dashboard look nicer
+    # Grafana expects timestamps with milliseconds
+    path = '/d/materialize-overview/materialize-overview'
+    query = urllib.parse.urlencode( {
+              "from": round(start.timestamp()) * 1000,
+                "to": round(dateparser.parse('in 30 seconds').timestamp()) * 1000,
```
Comment (ruchirK) [inline test/benchmark/avro-upsert/metric-verifier/wait_metric:41]: its not clear to me how to `"to"` timestamp depends on the end timestamp - I'm probably missing something and would love to know what

File: test/benchmark/avro-upsert/metric-verifier/wait_metric [hunk 2]
```diff
@@ -10,32 +10,63 @@
...
         # We aren't running query_range, so there should only be a single timestamp and point in the reponse
-        (ts, point) = current_values[0]['value']
-        if point == args.expected_value:
-            print(f"SUCCESS! Query settled in {time_so_far} seconds")
-            return 0
+        if current_values:
+            (ts, point) = [float(i) for i in current_values[0]['value']]
+            if point == args.expected_value:
+                rate = point / max(time_so_far, 1)
+                print(f"SUCCESS! Benchmark took {time_so_far}s, {rate} rows/sec")
```
Comment (ruchirK) [inline test/benchmark/avro-upsert/metric-verifier/wait_metric:68]: minor nit: would love for the rows / second to be rounded to the nearest row

File: test/benchmark/avro-upsert/README.md [hunk 3]
```diff
@@ -5,9 +5,30 @@ assumes that Kafka topics should have 30 partitions and that you want 16 worker
...
+run the version of this benchmark designed for smaller machines:
+
+    ./mzcompose run benchmark-medium
+
+## Looking at Performance
+
+Each benchmark will output something like the following:
+
+```sh
+SUCCESS! Benchmark finished in settled 766 seconds, 522193.21148825064 rows/sec
```
Comment (ruchirK) [inline test/benchmark/avro-upsert/README.md:29]: minor: It looks like the most recent version of the code says "Benchmark took <foo> seconds" but specifically "Benchmark finished in settled ..." jumped out as a small typo

**Review by ruchirK — APPROVED** [review]
LGTM!

File: test/benchmark/avro-upsert/metric-verifier/wait_metric (hunk 2 above)
Comment (cirego) [reply test/benchmark/avro-upsert/metric-verifier/wait_metric:68]: Sure thing!

File: test/benchmark/avro-upsert/README.md (hunk 3 above)
Comment (cirego) [reply test/benchmark/avro-upsert/README.md:29]: It's a typo! Will fix this up.

File: test/benchmark/avro-upsert/metric-verifier/wait_metric (hunk 1 above)
Comment (cirego) [reply test/benchmark/avro-upsert/metric-verifier/wait_metric:41]: Are you looking at an older version of the code? The current code calculates `to` without using an `end` timestamp.

Also, if we didn't supply a value for `to`, then the dashboard would default to an interval that may include other benchmarks.

Comment (cirego) [reply test/benchmark/avro-upsert/metric-verifier/wait_metric:41]: Oh, I see -- the comment mentions `end`. Will fix up!

**Pull request merged**